api/provider/aws/task_list.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+const MAX_TASK_IDS = 100
+
+func (t *TaskProvider) List() ([]models.TaskSummary, error) {
+	clusterNames, err := t.listClusterNames()
+	if err != nil {
+		return nil, err
+	}
+
+	taskCopies := map[string][]*ecs.Task{}
+	for _, clusterName := range clusterNames {
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
+
+		running := "RUNNING"
+		tasks, err := t.listTasksByStatus(&running, &environmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		stopped := "STOPPED"
+		stoppedTasks, err := t.listTasksByStatus(&stopped, &environmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		taskARNs := append(tasks, stoppedTasks...)
+		if len(taskARNs) > 0 {
+			describeTasksInput := &ecs.DescribeTasksInput{}
+			describeTasksInput.SetCluster(clusterName)
+
+			tasks, err := t.describeTasks(describeTasksInput)
+
+			if err != nil {
+				return nil, err
+			}
+
+			for _, task := range tasks {
+				if task.StartedBy != nil {
+					if strings.HasPrefix(*task.StartedBy, config.ENVVAR_INSTANCE) {
+						if _, ok := taskCopies[*task.StartedBy]; !ok {
+							taskCopies[*task.StartedBy] = []*ecs.Task{}
+						}
+
+						taskCopies[*task.StartedBy] = append(taskCopies[*task.StartedBy], task)
+					}
+				}
+			}
+		}
+	}
+
+	getModel := func(tasks []*ecs.Task) (*models.Task, error) {
```

Comment:
Seeing a lot of carryover from the old implementation. 

I think this "master" function should really only look something like this (psuedo code of course):

```
clusterNames, err := listClusterNames(...)

summaries := []*models.TaskSummary{}
for _, clusterName := range clusterNames {
    runningTaskARNs, err := listByStatus(clusterName, "RUNNING")
    stoppedTaskARNs, err := listByStatus(clusterName, "STOPPED")

    for _, taskARN := range append(runningTaskARNs, stoppedTaskARNs...) {
        summary := models.TaskSummary { ... }
        summaries = append(summaries, &summary)
    }
}

if err := t.populateTaskSummaries(summaries); err != nil { ... }

return summaries, nil
```